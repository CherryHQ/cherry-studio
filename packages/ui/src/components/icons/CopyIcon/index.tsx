// Original path: src/renderer/src/components/Icons/CopyIcon.tsx
import { Copy } from 'lucide-react'

const CopyIcon = (props: React.ComponentProps<typeof Copy>) => <Copy size="1rem" {...props} />

export default CopyIcon
