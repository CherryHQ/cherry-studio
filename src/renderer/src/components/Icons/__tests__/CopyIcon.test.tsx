import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import CopyIcon from '../CopyIcon'

describe('CopyIcon', () => {
  it('should render with default class', () => {
    const { container } = render(<CopyIcon />)
    const icon = container.querySelector('i')

    expect(icon).toBeInTheDocument()
    expect(icon).toHaveClass('iconfont')
    expect(icon).toHaveClass('icon-copy')
  })

  it('should merge custom className with default classes', () => {
    const customClass = 'custom-icon-class'
    const { container } = render(<CopyIcon className={customClass} />)
    const icon = container.querySelector('i')

    expect(icon).toHaveClass('iconfont')
    expect(icon).toHaveClass('icon-copy')
    expect(icon).toHaveClass(customClass)
  })

  it('should pass through additional props', () => {
    const onClick = vi.fn()
    const { container } = render(<CopyIcon onClick={onClick} title="Copy to clipboard" data-testid="copy-icon" />)
    const icon = container.querySelector('i')

    expect(icon).toHaveAttribute('title', 'Copy to clipboard')
    expect(icon).toHaveAttribute('data-testid', 'copy-icon')

    icon?.click()
    expect(onClick).toHaveBeenCalledTimes(1)
  })

  it('should match snapshot', () => {
    const { container } = render(<CopyIcon />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
