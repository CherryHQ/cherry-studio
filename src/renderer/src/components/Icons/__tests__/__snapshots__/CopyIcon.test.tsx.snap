// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyIcon > should match snapshot 1`] = `
<i
  class="iconfont icon-copy undefined"
/>
`;
